apis/github/code/src/tools.js

```diff
@@ -0,0 +1,214 @@
+export async function closeIssue(octokit, owner, repo, issueNumber) {
+    try {
+        await octokit.issues.update({
+            owner,
+            repo,
+            issue_number: issueNumber,
+            state: 'closed'
+        });
+        console.log(`Closed issue #${issueNumber} - https://github.com/${owner}/${repo}/issues/${issueNumber}`);
+    } catch (error) {
+        console.error(`Error closing issue: ${error}`);
+    }
+}
+
+export async function listRepos(octokit, owner) {
+    try {
+        const repos = await octokit.repos.listForUser({
+            username: owner,
+            per_page: 100
+        });
+
+        repos.data.forEach(repo => {
+            console.log(`${repo.name} (ID: ${repo.id}) - https://github.com/${owner}/${repo.name}`);
+        });
+    } catch (error) {
+        console.error(`Error listing repositories: ${error}`);
+    }
+}
+
+export async function listIssues(octokit, owner, repo) {
+    const issues = await octokit.issues.listForRepo({
```

Comment:
Many of the functions starting from this point don't have the try-catch setup that the earlier ones did. Is it something that is needed?

Reply:
no, catching and logging at the top-level now